Tests/TensorFlowTests/OptimizerTests.swift

```diff
@@ -0,0 +1,54 @@
+// Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import XCTest
+@testable import TensorFlow
+
+final class OptimizerTests: XCTestCase {
+    func testRAdam() {
+        struct Model: Layer {
+            var w: Tensor<Float>
+            var b: Tensor<Float>
+
+            public init(w: Tensor<Float>, b: Tensor<Float>) {
+                self.w = w
+                self.b = b
+            }
+
+            @differentiable
+            public func callAsFunction(_ input: Tensor<Float>) -> Tensor<Float> {
+                return matmul(input, w) + b
+            }
+        }
+        let w = Tensor<Float>([[1.0], [2.0]])
+        let b = Tensor<Float>([0.0])
+        var model = Model(w: w, b: b)
+        let optimizer = RAdam(for: model, learningRate: 1e-3)
+        // To obtain gradient of type Model.TangentVector 
+        var gradient = model.gradient { model -> Tensor<Float> in 
+            return meanAbsoluteError(predicted: model(w.tranposed()), expected: Tensor<Float>(1.0))
+        }
+        // Custom gradient passed to check optimizer validity
+        gradient.w = Tensor<Float>([[0.1], [0.2]])
+        for _ in 0..<1000 {
+            optimizer.update(&model, along: gradient)
+        }
+        let expected_gradient = Tensor<Float>([[0.5543607], [1.55286]])
```

Comment:
Where did this value come from? Could you please share a reference implementation (e.g. from Python)? This seems very important for this change.

Reply:
https://github.com/tensorflow/addons/blob/b4cea8189724f88663637daa1df82d035d97cba4/tensorflow_addons/optimizers/rectified_adam_test.py#L73

The values aren't exactly the same, but can be(?) approximated I suppose.

Reply:
@dan-zheng official pytorch implementation [here](https://github.com/LiyuanLucasLiu/RAdam/blob/master/radam/radam.py).

Reply:
Unfortunately the official PyTorch version did not come with any value based tests, so I had to take the tests from the tensorflow/addons implementation

Reply:
It doesn't matter where the reference implementation comes from, but checking it in code is important! Some shorter reference implementations are included in inline comments, others are checked in as Python files under [`Tools/ReferenceImplementations`](https://github.com/tensorflow/swift-apis/tree/master/Tools/ReferenceImplementations).

Could you please check in a reference implementation in code? When run, the reference implementation should print out the ~exact values used in the test.

Reply:
I have checked these values in a Colab notebook. Here's the gist: https://gist.github.com/vballoli/8f882fd2b5bbafbf98ff28d945ad2f2f#file-blank_swift-ipynb

Reply:
I've updated the GItHub gist link.

Reply:
Perhaps you misunderstood: I'm asking if you could please add or link a Python reference implementation (expected values) that checks the Swift RAdam implementation tweaks here (actual values). Does that make sense?

In other words: how do you know `let expected_gradient = Tensor<Float>([[0.5543607], [1.55286]])` is the expected gradient? Did you run some Python RAdam program to get those values? If yes, please add/link the Python code!

Reply:
@vballoli, Similarly to layer gradient tests, For example [Conv1D gradient](https://github.com/tensorflow/swift-apis/blob/c3f38deb27932ee38af551c3e44738083c25b37d/Tests/TensorFlowTests/LayerTests.swift#L138), you have to run the pytorch/tensorflow code and add the equivalent code in the test in a similar format. Both the test and the reference code should result in the same output.

Reply:
Ah, I get it. Will work on that. I found there was an issue with RAdam since accuracy had dropped and training loss increased drastically after 50 epochs or so. So when I went back to the source code, I found the missing factor that wasn't multiplied.